R/plotPValHist.R

```diff
@@ -60,50 +59,80 @@ plotPvalHist <- function(P.Val,
                              times         = SampNames,
                              new.row.names = sequence(prod(length(SampNames), nrow(P.Val))))
 
-    if (facet) {
-        numcol <- 3
-        numrow <- (NumSamples / numcol) %>% ceiling
-
-        Hist_Pval_Facet <- ggplot2::ggplot(data = P.Val, aes(x = Pval)) +
-            ggplot2::geom_histogram(alpha = alpha,
-                                    fill = fill,
-                                    color = color,
-                                    binwidth = binWidth) +
-            ggplot2::xlab("P-value") +
-            ggplot2::ylab("Count") +
-            ggtitle("P-value Histograms") +
-            ggplot2::scale_fill_brewer(palette = "Set1") +
-            ggplot2::facet_wrap(~Levels, nrow = numrow, scales = "free") +
-            theme_grey() + baseTheme(baseFontSize)
+    title <- "P-value Histograms"
+    plotlist <- list()
+    if (plotType == "canvasXpress") {
+        if (facet) {
+            cx.data <- subset(P.Val, select = Pval)
+            var.annot <- subset(P.Val, select = -c(Pval))
+            plotlist <- canvasXpress::canvasXpress(data                    = cx.data,
+                                                   varAnnot                = var.annot,
+                                                   histogramData           = TRUE,
+                                                   graphType               = "Scatter2D",
+                                                   histogramBins           = binWidth,
+                                                   colors                  = color,
+                                                   title                   = title,
+                                                   xAxisTitle              = "P-value",
+                                                   yAxisTitle              = "Count",
+                                                   hideHistogram           = FALSE,
+                                                   showHistogramDensity    = FALSE,
+                                                   showLegend              = FALSE,
+                                                   segregateVariablesBy    = list("Levels"))
+        } else {
+            for (i in 1:NumSamples) {
+                s <- SampNames[i]
+                MyPVal <- dplyr::filter(P.Val, grepl(s, Levels))
```

Comment:
Why using dplyr::filter and base subset at the same time?

Reply:
I am using subset function to select and deselect columns. otherwise, I have to import dplyr::select